firebase-installations/src/androidTest/java/com/google/firebase/installations/FirebaseInstallationsInstrumentedTest.java

```diff
@@ -26,4 +47,99 @@
  */
 @RunWith(AndroidJUnit4.class)
 @FixMethodOrder(MethodSorters.NAME_ASCENDING)
-public class FirebaseInstallationsInstrumentedTest {}
+public class FirebaseInstallationsInstrumentedTest {
+  private FirebaseApp firebaseApp;
+  @Mock private FirebaseInstallationServiceClient backendClientReturnsOk;
+  @Mock private FirebaseInstallationServiceClient backendClientReturnsError;
+  private FiidCache actualCache;
+  @Mock private FiidCache cacheReturnsError;
+
+  @Before
+  public void setUp() throws FirebaseInstallationServiceException {
+    MockitoAnnotations.initMocks(this);
+    FirebaseApp.clearInstancesForTest();
+    firebaseApp =
+        FirebaseApp.initializeApp(
+            ApplicationProvider.getApplicationContext(),
+            new FirebaseOptions.Builder()
+                .setApplicationId("1:123456789:android:abcdef")
+                .setProjectId("project-id")
+                .setApiKey("api_key")
+                .build());
+    actualCache = new FiidCache(firebaseApp);
+    when(backendClientReturnsOk.createFirebaseInstallation(
+            anyString(), anyString(), anyString(), anyString()))
+        .thenReturn(
+            InstallationResponse.builder()
+                .setName("/projects/123456789/installations/fid")
+                .setRefreshToken("refresh-token")
+                .setAuthToken(
+                    InstallationTokenResult.builder()
+                        .setToken("auth-token")
+                        .setTokenExpirationTimestampMillis(1000L)
+                        .build())
+                .build());
+    when(backendClientReturnsError.createFirebaseInstallation(
+            anyString(), anyString(), anyString(), anyString()))
+        .thenThrow(
+            new FirebaseInstallationServiceException(
+                "SDK Error", FirebaseInstallationServiceException.Status.SERVER_ERROR));
+    when(cacheReturnsError.insertOrUpdateCacheEntry(any())).thenReturn(false);
+    when(cacheReturnsError.readCacheEntryValue()).thenReturn(null);
+  }
+
+  @After
+  public void cleanUp() throws Exception {
+    actualCache.clear();
+  }
+
+  @Test
+  public void testCreateFirebaseInstallation_CacheOk_BackendOk() throws Exception {
+    FirebaseInstallations firebaseInstallations =
+        new FirebaseInstallations(firebaseApp, actualCache, backendClientReturnsOk);
+
+    // No exception, means success.
+    assertThat(Tasks.await(firebaseInstallations.getId())).isNotEmpty();
+    FiidCacheEntryValue entryValue = actualCache.readCacheEntryValue();
```

Comment:
Ok, I have a few problems here:
a) What is Fiid? It's called FID.
b) I'm a bit worried about the term "cache" here.
As a cache I understand something that is temporarily in memory of a system in order to safe multiple database- or filesystem-lookups or requests, but when the system shuts down the cache is usually deleted.

Isn't this the local storage, where the FID (and other data) gets persisted?
If yes, can't we call is something like that, e.g. "FidStorageEntry" or "PersistedFidEntry", I don't know.

Happy to discuss in person.